src/index.ts

```diff
@@ -225,23 +225,31 @@ async function generateDocumentation(hre: HardhatRuntimeEnvironment): Promise<vo
     ) => {
       if ('documentation' in astNode === false) return;
 
-      const paramDoc = astNode.documentation.text
+      const paramDocs = astNode.documentation.text
         .match(/@.*/g)
         .filter((text: string) => text.match(/@param.*/));
 
-      if (paramDoc.length > 0) {
-        astNode.parameters.parameters.forEach((param: any, index: number) => {
-          // Check if there is not the same number of Natspec @param tags compared
-          // to the number of params for the function in the AST node.
-          if (paramDoc[index] === undefined) return;
-
-          const paramName = param.name;
-          const paramType = param.typeDescriptions.typeString;
-
-          doc[docEntry][functionName].inputs[paramName] = {
-            type: paramType,
-            description: paramDoc[index].replace(`@param ${paramName} `, ''),
-          };
+      if (paramDocs.length > 0) {
+        astNode.parameters.parameters.forEach((param: any) => {
+          paramDocs.forEach((paramDoc: any) => {
+            if (paramDoc.replace('@param ', '').startsWith(param.name)) {
+              const paramName = param.name;
+              const paramType = param.typeDescriptions.typeString;
```

Comment:
Maybe you can put these 2 constants above the `if` statement, as they are repeated in both the `if` and `else` conditional.

Reply:
true